src/network/network.cpp

```diff
@@ -841,6 +841,27 @@ static void NetworkInitGameInfo()
 	strecpy(ci->client_name, _settings_client.network.client_name, lastof(ci->client_name));
 }
 
+/**
+ * Check whether the client and server name are set, for a dedicated server and if not set them to some default
+ * value and tell the user to change this as soon as possible.
+ * If the saved name is the default value, then the user is told to override  this value too.
+ * This is only meant dedicated servers, as for the other servers the GUI ensures a name has been entered.
+ */
+static void CheckClientAndServerName()
+{
+	static const char *fallback_client_name = "Unnamed Client";
+	if (StrEmpty(_settings_client.network.client_name) || strcmp(_settings_client.network.client_name, fallback_client_name) == 0) {
+		DEBUG(net, 0, "No \"client_name\" has been set, using \"%s\" instead. Please set this now using the \"name <new name>\" command.", fallback_client_name);
+		strecpy(_settings_client.network.client_name, fallback_client_name, lastof(_settings_client.network.client_name));
+	}
+
+	static const char *fallback_server_name = "Unnamed Server";
```

Comment:
Maybe we can use `<client_name>'s Server` as fallback? But maybe not for this PR (2 more places that could use that)

Reply:
In the UI that could indeed be a decent default. For the dedicated server I'm fairly server you'll then usually get "Unnamed Client's Server" as server name.
The other place where the "Unnamed Server" is set now should go away once the pre-validation of string settings has been implemented.